translate/tools/test_pocount.py

```diff
@@ -169,14 +179,61 @@ def test_totalsourcewords(self):
 @mark.parametrize("style", ["csv", "full", "short-strings", "short-words"])
 @mark.parametrize("incomplete", [True, False], ids=lambda v: f"incomplete={v}")
 @mark.parametrize("no_color", [True, False], ids=lambda v: f"no-color={v}")
-def test_output(style, incomplete, no_color, capsys, snapshot):
+def test_output(style, incomplete, no_color, capsys: CaptureFixture[str], snapshot):
     opts = [f"--{style}"]
     if incomplete:
         opts.append("--incomplete")
     if no_color:
         opts.append("--no-color")
 
-    pocount.main([*opts, *test_po_files])
+    pocount.main([*opts, *test_files])
     stdout = capsys.readouterr()[0]
 
     assert stdout == snapshot
+
+
+@mark.parametrize(
+    "opts",
+    [
+        param([_po_file, "--no-color"], id="po-file"),
+        param([_po_fuzzy, "--no-color"], id="po-file-fuzzy"),
+        param([_po_csv, "--no-color", "--csv"], id="po-file-csv"),
+        param([_xliff_states_yes, "--no-color"], id="xliff-states-yes"),
+        param([_xliff_states_no, "--no-color"], id="xliff-states-no"),
+    ],
+)
+def test_cases(opts, capsys: CaptureFixture[str], snapshot):
+    pocount.main(opts)
+
+    result = capsys.readouterr()
+
+    assert result == snapshot
+
+
+@mark.parametrize(
+    "opts,expected",
+    [
+        param(
+            ["--csv", "--short"],
+            "argument --short: not allowed with argument --csv",
+            id="mutually-exclusive",
+        ),
+        param(
+            ["missing.po"],
+            "cannot process missing.po: does not exist",
+            id="missing-file",
+        ),
+        param([], "the following arguments are required: files", id="no-args"),
+    ],
+)
+def test_error_cases(opts, expected):
+    # We're using special case for this, to produce correct output.
+    # Also, using partial matching instead of snapshots, becouse mac-os argparse
+    # output is slightly different.
+    result = subprocess.run(
+        [sys.executable, pocount.__file__, *opts],
+        capture_output=True,
+        text=True,
+    )
```

Comment:
The code executed this way will not be accounted in coverage. The existing functional tests do that via the `PYTHON_ARGS` environment passed to the python interpreter.

Reply:
Ah nice. I didn't look at the coverage report, only at the code. I didn't consider the option that pytest might be handling that automatically.